src/scaler/manager.rs

```diff
@@ -724,3 +733,72 @@ where
     }
     scalers
 }
+
+/// Based on the name of the model and the optionally provided ID, returns a unique ID for the
+/// component that is a sanitized version of the component reference and model name, separated
+/// by a dash.
+pub(crate) fn component_id(
+    model_name: &str,
+    component_id: Option<&String>,
```

Comment:
If someone manually specified a component ID, wouldn't we want to use that _instead_ of generating one for them?

Reply:
Yes! That's what this does, and we have a test to assert it too.